modules/gdscript/gdscript_editor.cpp

```diff
@@ -1710,11 +1710,19 @@ static bool _guess_expression_type(GDScriptParser::CompletionContext &p_context,
 						GDScriptParser::DataType native_type = base.type;
 
 						while (native_type.kind == GDScriptParser::DataType::CLASS) {
-							native_type = native_type.class_type->base_type;
+							if (native_type.is_meta_type) {
```

Comment:
I would personally put a comment or two explaining why these changes are done, pointing to the original issue (Similar comments say e.g. "`GH-57669`" across the codebase).

Reply:
I don't really think a comment is needed here. It's quite self explanatory if you know what `is_meta_type` means. And meta types aren't like specific to this issue. They should be handled properly anyway.